Jenkinsfile

```diff
@@ -0,0 +1,169 @@
+    fieldsObject.put('value', branchName.toString());
+    fieldsObject.put('short', true);
+
+    fields.add(fieldsObject); 
+
+    fieldsObject.put('title','Status'); 
+    fieldsObject.put('value',buildStatus.toString()); 
+    fieldsObject.put('short',true); 
+    
+    fields.add(fieldsObject); 
+    
+    fieldsObject = new JSONObject(); 
+    fieldsObject.put('title','Job ID'); 
+    fieldsObject.put('value','#' + env.BUILD_NUMBER.toString()); 
+    fieldsObject.put('short',true); 
+    
+    fields.add(fieldsObject); 
+    
+    // Put fields JSONArray 
+    detailsAttachment.put('pretext',"Ran DeviceDB CI pipeline on Jenkins"); 
+    detailsAttachment.put('title',"Sonarqube Dashboard");
+    detailsAttachment.put('title_link',"http://pe-jm.usa.arm.com:9000/dashboard?id=edge%3Adevicedb%3A"+"${branchName}"); 
+    //detailsAttachment.put('author_name',"LAVA Job");
+    //detailsAttachment.put('author_link',"http://lava.mbedcloudtesting.com/scheduler/alljobs");
+    detailsAttachment.put('text',"Click to view Sonarqube Dashboard");
+    detailsAttachment.put('fields',fields); 
+    detailsAttachment.put('color', colorCode.toString());
+    detailsAttachment.put('footer','After ' + Util.getTimeSpanString(System.currentTimeMillis() - currentBuild.startTimeInMillis)) 
+    
+    attachments.add(detailsAttachment); 
+
+    print detailsAttachment
+    
+    // Send notifications 
+    slackSend (message: mainText.toString(), attachments: attachments.toString()) 
+    
+}
+
+
+pipeline {
+  agent none
+  options{
+    skipDefaultCheckout()
+  }
+  stages {
+    stage('Build') {
+        agent{
+            label 'noi-linux-ubuntu16-ci-slave'
+        }
+        options{
+            lock resource: 'noi-linux-ubuntu16-ci-slave'
+        }
+        steps {
+         //checkout scm
+          withEnv(["GOROOT=/home/jenkins/go", "GOPATH=/home/jenkins/goprojects", "PATH+GO=/home/jenkins/goprojects/bin:/home/jenkins/go/bin"]){
+              script{
+                  sh 'rm -rf /home/jenknis/goprojects/src/github.com/armPelionEdge/devicedb'
+                  if(env.BRANCH_NAME ==~ /^PR-[0-9]*$/){
+                    sh "go get -u github.com/armPelionEdge/devicedb && cd /home/jenkins/goprojects/src/github.com/armPelionEdge/devicedb && git fetch --all && git checkout ${CHANGE_BRANCH} && git merge origin/${env.CHANGE_TARGET} && go build"
+                  }
+                  else{
+                    sh "go get -u github.com/armPelionEdge/devicedb && cd /home/jenkins/goprojects/src/github.com/armPelionEdge/devicedb && git fetch --all && git checkout ${env.BRANCH_NAME} && git merge origin/${env.BRANCH_NAME} && go build"
+                  }
+              }
+          }
+       }
+    }
+    
+    //stage('Test and Code Review') {
+      //parallel {
+        stage('Test'){
+          agent{
+            label 'noi-linux-ubuntu16-ci-slave'
+          }
+          steps {
+            catchError(buildResult: 'SUCCESS'){
```

Comment:
What's the reason that this is here?

Reply:
As far as I remember that was required for devicejs but while I was copying certain things I missed removing this part.

Reply:
Removed that line. It was not necessary in this case.